neo/public/javascripts/map/controllers.js

```diff
@@ -338,29 +341,32 @@ angular.module('cloudberry.map', ['leaflet-directive', 'cloudberry.common'])
 
     function loadCityJsonByBound(onEachFeature){
       var bounds = $scope.map.getBounds();
-      var rteBounds = "city/" + bounds._northEast.lat + "/" + bounds._southWest.lat + "/" + bounds._northEast.lng + "/" + bounds._southWest.lng;
-      $http.get(rteBounds)
-        .success(function(data) {
-          $scope.geojsonData.city = data;
+
+      var data_from_cache = Cache.getCityPolygonsFromCache(bounds).done(function(data_response){
+
+          $scope.geojsonData.city = data_response;
+
           if($scope.polygons.cityPolygons) {
-            $scope.map.removeLayer($scope.polygons.cityPolygons);
-          }
-          $scope.polygons.cityPolygons = L.geoJson(data, {
-            style: $scope.styles.cityStyle,
-            onEachFeature: onEachFeature
-          });
-          setCenterAndBoundry($scope.geojsonData.city.features);
+                $scope.map.removeLayer($scope.polygons.cityPolygons);
+              }
+          $scope.polygons.cityPolygons = L.geoJson(data_response, {
+                            style: $scope.styles.cityStyle,
+                            onEachFeature: onEachFeature
+              });
+
+
```

Comment:
General comments: Looks like the only change between the line 346 and 352 is data -> data_response. Can you maintain the original format (indent) to reflect your change only?